guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/SharedInjectorsTest.java

```diff
@@ -0,0 +1,209 @@
+package name.falgout.jeffrey.testing.junit.guice;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.name.Names;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.inject.Named;
+import name.falgout.jeffrey.testing.junit.guice.SharedInjectorsTest.OuterClassModule;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(GuiceExtension.class)
+@IncludeModule(OuterClassModule.class)
+@SharedInjectors
+class SharedInjectorsTest {
+
+  private static final Random RANDOM = new Random();
+
+  private static String getRandomString() {
+    return String.valueOf(RANDOM.nextInt(Integer.MAX_VALUE));
+  }
+
+  @Test
+  @IncludeModule(TestCaseModule.class)
+  @SharedInjectors
+  void test1(@Named("base") String base, @Named("test") String test) {
+    assertNotNull(base);
```

Comment:
Can you check the actual value of the strings? Use deterministic strings instead of random ones